cmd/sqlcmd/main.go

```diff
@@ -201,22 +201,27 @@ func setConnect(connect *sqlcmd.ConnectSettings, args *SQLCmdArguments, vars *sq
 	connect.ErrorSeverityLevel = args.ErrorSeverityLevel
 }
 
+type allocConsole func(historyFile string) sqlcmd.Console
+
+var consoleAllocator allocConsole = console.NewConsole
+
 func run(vars *sqlcmd.Variables, args *SQLCmdArguments) (int, error) {
 	wd, err := os.Getwd()
 	if err != nil {
 		return 1, err
 	}
 
-	iactive := args.InputFile == nil && args.Query == ""
+	var connectConfig sqlcmd.ConnectSettings
+	setConnect(&connectConfig, args, vars)
 	var line sqlcmd.Console = nil
-	if iactive {
-		line = console.NewConsole("")
+	if connectConfig.RequiresPassword() {

```

Comment:
don't you still need the iactive check? How are you testing this before pushing?

Reply:
My bad. I missed that locally it is only using SQL authentication.
Adding testcases for AAD modes

Reply:
Added the iactive check.